pkg/config/v1alpha5/types.go

```diff
@@ -145,6 +145,13 @@ type SimpleConfigRegistries struct {
 	Config string                            `mapstructure:"config" json:"config,omitempty"` // registries.yaml (k3s config for containerd registry override)
 }
 
+type SimpleConfigFile struct {
+	Path        string `mapstructure:"path" json:"path,omitempty"`
```

Comment:
Could we just use `source` and merge `path` and `content`?
Or do you see any edge case where a single line path would be supposed to be the content of the file?
This would also help with the JSONSchema issue you're seeing.

Reply:
> Could we just use `source` and merge `path` and `content`? Or do you see any edge case where a single line path would be supposed to be the content of the file? This would also help with the JSONSchema issue you're seeing.

My concern is that the logic to determine if `source` is a file or not across different operating systems and file systems will be fragile.

We will have 3 cases:
- If the input is a valid file path and it exists.
- If the input is a valid file path and it doesn't exist.
- If the input is the actual content.

To differentiate between the 2nd and 3rd case I don't see a sold way to do that :thinking:

Reply:
With yaml.v3 you should be able to distinguish between

```yaml
source: /path/treated/as/path.yaml
```
and
```yaml
source: |
	/path/not/treated/as/pathyaml
```

So a single line string as in the first example, which is also a valid path, will be treated as a path (existent or not) - everything else won't.
I.e. a non-path-like string or a `|` or `>` (or whatever else there is) denoted string won't be treated as a path but as file content.

WDYT?

Reply:
@iwilltry42 Thanks for the hint, I will give it a try :ok_hand:

Reply:
See https://pkg.go.dev/gopkg.in/yaml.v3#Style
and the explanations, e.g. at https://redocly.com/docs/yaml/blocks-and-flows/
-> So basically we decide between scalar and block scalars (I wouldn't support flow scalars here).